web-auth/src/components/Auth.svelte

```diff
@@ -20,111 +19,264 @@
 
   let email = "";
   let step: AuthStep = AuthStep.Base;
   let webAuth: WebAuth;
+  let isExistingUser = false;
+  let tokenExpiry = 0;
+  let managementApiToken = null;
 
-  $: isLegacy = false;
+  $: errorText = "";
+  $: isAllowedClient = false;
+  $: domainDisabled = isDomainDisabled(email);
+  $: headingText = getHeadingText(step, isExistingUser, email);
+  $: subheadingText = getSubheadingText(step, email);
 
   function isDomainDisabled(email: string): boolean {
     return disableForgotPassDomainsArr.some((domain) =>
       email.toLowerCase().endsWith(domain.toLowerCase()),
     );
   }
 
-  $: domainDisabled = isDomainDisabled(email);
+  function configureDevMode() {
+    if (
+      import.meta.env.DEV &&
+      (!configParams || configParams === "undefined")
+    ) {
+      console.warn(
+        "No auth config provided. In development mode - auth flows will not work.",
+      );
+      errorText = "Authentication is not configured in development mode";
+      isAllowedClient = true;
+
+      step = AuthStep.Base;
+      return;
+    }
+  }
+
+  function init() {
+    try {
+      configureDevMode();
+
+      const config = JSON.parse(
+        decodeURIComponent(escape(window.atob(configParams))),
+      ) as Config;
+
+      if (!cloudClientIDsArr.includes(config?.clientID)) {
+        errorText = "Authentication is not available for this client";
+        isAllowedClient = false;
+        return;
+      }
+      isAllowedClient = true;
 
-  function initConfig() {
-    const config = JSON.parse(
-      decodeURIComponent(escape(window.atob(configParams))),
-    ) as Config;
+      const isSignup = config?.extraParams?.screen_hint === "signup";
 
-    const isSignup = config?.extraParams?.screen_hint === "signup";
+      if (isSignup) {
+        step = AuthStep.SignUp;
+      }
 
-    if (isSignup) {
-      step = AuthStep.SignUp;
+      const authOptions: AuthOptions = Object.assign(
+        {
+          overrides: {
+            __tenant: config.auth0Tenant,
+            __token_issuer: config.authorizationServer.issuer,
+          },
+          domain: config.auth0Domain,
+          clientID: config.clientID,
+          redirectUri: config.callbackURL,
+          responseType: "code",
+        },
+        config.internalOptions,
+      );
+
+      webAuth = new auth0.WebAuth(authOptions);
+    } catch (e) {
+      console.error("Failed to initialize auth:", e);
+      errorText = "Failed to initialize authentication in development mode";
+    }
+  }
+
+  function authorizeSSO(email: string, connectionName: string) {
+    if (import.meta.env.DEV) {
+      errorText = "SSO authentication is not available in development mode";
+      return;
     }
 
-    if (cloudClientIDsArr.includes(config?.clientID)) {
-      isLegacy = true;
+    webAuth.authorize({
+      connection: connectionName,
+      login_hint: email,
+      prompt: "login",
+    });
+  }
+
+  // TODO: Serverless function
```

Comment:
Where can we deploy this token generation function on the server side in our current techstack? @ericpgreen2

https://www.notion.so/rilldata/Post-Canvas-Implementation-Notes-190ba33c8f5780c99e8bd66bda682125#192ba33c8f5780c4b4ebf73797b96b49

Reply:
Where can I deploy this piece of server-side code? Primarily the usage is for generating bearer token and checking user existence @begelundmuller

Reply:
What about in an Auth0 Action? IIRC, that's what our Auth0 contact advised.

Reply:
Explained in https://www.notion.so/rilldata/Post-Canvas-Implementation-Notes-190ba33c8f5780c99e8bd66bda682125

Reply:
So, one of Auth0's recommendations was to move HRD from client-side to the Auth0-side. If we did that, and let's say HRD doesn't recognize an email – does that help us?

Reply:
If HRD doesn't recognize an email, the user will have to switch to the sign-up form to create a new account. HRD is only useful for Enterprise and Social logins based on the provided user email.

If we want to ensure a unified auth flow, we need to add a user existence check hence https://rilldata.slack.com/archives/C080SHAMZU2/p1738953939372419